apps/dashboard/src/components/workflow-editor/steps/skip-conditions-button.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button } from '@/components/primitives/button';
+import { Separator } from '@/components/primitives/separator';
+import { SidebarContent } from '@/components/side-navigation/sidebar';
+import { useFeatureFlag } from '@/hooks/use-feature-flag';
+import { FeatureFlagsKeysEnum, StepResponseDto } from '@novu/shared';
+import { useMemo } from 'react';
+import { RiArrowRightSLine, RiGuideFill } from 'react-icons/ri';
+import { RQBJsonLogic } from 'react-querybuilder';
+import { parseJsonLogic } from 'react-querybuilder/parseJsonLogic';
+import { Link } from 'react-router-dom';
+
+export function SkipConditionsButton({ step, inSidebar = false }: { step: StepResponseDto; inSidebar?: boolean }) {
+  const isStepConditionsEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_STEP_CONDITIONS_ENABLED);
+  const uiSchema = step.controls.uiSchema;
+  const skip = uiSchema?.properties?.skip;
+
+  const conditionsCount = useMemo(() => {
+    if (!step.controls.values.skip) return 0;
+
+    const query = parseJsonLogic(step.controls.values.skip as RQBJsonLogic);
+
+    return query.rules.length;
+  }, [step]);
+
+  const button = (
+    <Link to={'./conditions'} relative="path" state={{ stepType: step.type }}>
+      <Button variant="secondary" mode="outline" className="flex w-full justify-start gap-1.5 text-xs font-medium">
+        <RiGuideFill className="h-4 w-4 text-neutral-600" />
+        Skip Conditions
+        <span className="ml-auto flex items-center gap-0.5">
+          <span>{conditionsCount}</span>
+          <RiArrowRightSLine className="ml-auto h-4 w-4 text-neutral-600" />
+        </span>
+      </Button>
+    </Link>
+  );
+
+  if (!skip || !isStepConditionsEnabled) {
+    return null;
+  }
+
+  if (!inSidebar) {
+    return button;
+  }
+
+  return (
+    <>
+      <SidebarContent>{button}</SidebarContent>
+      <Separator />
```

Comment:
feel to me that the separator is the responsibility of the parent

Reply:
I explained it here a bit: https://github.com/novuhq/novu/pull/7592/files/6c796518c845186a2b8445177502bddf8e58a628#r1930496411

It relates to the parent, but at the same time, both `Separator` and `SidebarContent` needs to be rendered only if this child component is rendered.

Either you put everything to the parent, so this component wouldn't need to exist and all logic related **only** to this button would live in parent - which already has 400 lines.

Or you do it like I did here by using `inSidebar` flag, which essentially just wraps the component so it can be shown in sidebar - better way could be to create a wrapper component for all sidebar items, but that would be for separate PR.

Do you have some other suggestion how we could solve this?

Reply:
https://github.com/novuhq/novu/pull/7592/files#r1931907899